tests/smc/from_blackjax/test_blackjax_smc.py

```diff
@@ -0,0 +1,173 @@
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+import chex
+import jax
+import numpy as np
+import pytest
+import scipy
+
+from numpy import dtype
+from xarray.core.utils import Frozen
+
+import pymc as pm
+
+from pymc import to_inference_data
+from pymc.backends.base import MultiTrace
+from pymc.smc.from_blackjax.sampling import (
+    arviz_from_particles,
+    blackjax_particles_from_pymc_population,
+    get_jaxified_loglikelihood,
+    get_jaxified_logprior,
+    sample_with_blackjax_smc,
+)
+from tests.smc.conftest import fast_model, two_gaussians_model
+
+
+@pytest.mark.parametrize(
+    "kernel, check_for_integration_steps, inner_kernel_params",
+    [
+        ("HMC", True, {"step_size": 0.1, "integration_steps": 11}),
+        ("NUTS", False, {"step_size": 0.1}),
+    ],
+)
+def test_blackjax_smc(kernel, check_for_integration_steps, inner_kernel_params):
+    """
+    When running the two gaussians model
+    with BlackJax SMC, we sample them correctly,
+    the shape of a posterior variable is (1, particles, dimension)
+    and the inference_data has the right attributes.
+
+    """
+    model, muref = two_gaussians_model()
+    iterations_to_diagnose = 2
+    n_particles = 1000
+    with model:
+        inference_data = sample_with_blackjax_smc(
+            n_particles=n_particles,
+            kernel=kernel,
+            inner_kernel_params=inner_kernel_params,
+            iterations_to_diagnose=iterations_to_diagnose,
+        )
+
+    x = inference_data.posterior["X"]
+
+    assert x.to_numpy().shape == (1, n_particles, 4)
+    mu1d = np.abs(x).mean(axis=0).mean(axis=0)
+    np.testing.assert_allclose(muref, mu1d, rtol=0.0, atol=0.03)
+
+    for attribute, value in [
+        ("particles", n_particles),
+        ("step_size", 0.1),
+        ("num_mcmc_steps", 10),
+        ("iterations_to_diagnose", iterations_to_diagnose),
+        ("sampler", f"Blackjax SMC with {kernel} kernel"),
+    ]:
+        assert inference_data.posterior.attrs[attribute] == value
+
+    for diagnostic in ["lambda_evolution", "log_likelihood_increments"]:
+        assert inference_data.posterior.attrs[diagnostic].shape == (iterations_to_diagnose,)
+
+    for diagnostic in ["ancestors_evolution", "weights_evolution"]:
+        assert inference_data.posterior.attrs[diagnostic].shape == (
+            iterations_to_diagnose,
+            n_particles,
+        )
+
+    for attribute in ["running_time_seconds", "iterations"]:
+        assert attribute in inference_data.posterior.attrs
+
+    if check_for_integration_steps:
+        assert inference_data.posterior.attrs["integration_steps"] == 11
+
+
+def test_blackjax_particles_from_pymc_population_univariate():
+    model = fast_model()
+    population = {"x": np.array([2, 3, 4])}
+    blackjax_particles = blackjax_particles_from_pymc_population(model, population)
+    chex.assert_trees_all_close(blackjax_particles, [np.array([[2], [3], [4]])])
```

Comment:
I dont think PyMC use chex - use `jax.tree_map(np.testing.assert..., ...)` instead

Reply:
the dependency is transitive (pymc -> blackjax -> optax -> chex)